### PR TITLE
feat(Intersections): inspect intersection between TriangulatedSurface…

### PR DESCRIPTION
**Pull request by panquez**
…s and EdgedCurves

### CODE REVIEW

**Review by BotellaA — APPROVED** [review]

File: src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+                const auto result_colinear =
+                    geode::colinear_segment_segment_intersection_detection(
+                        segment, edge );
+                if( result_colinear.first != geode::Position::outside
+                    || result_colinear.second != geode::Position::outside )
+                {
+                    return true;
+                }
+            }
+        }
```
Comment (MelchiorSchuh) [inline src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp:114]: You can add a if( edge.pt1 or edge.pt2 is inside the triangle ) return true

File: src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+        std::vector< std::pair< geode::index_t, geode::index_t > >
+            intersecting_elements()
+        {
+            return std::move( intersecting_elements_ );
+        }
+
+    private:
+        bool triangle_edge_intersect(
```
Comment (MelchiorSchuh) [inline src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp:71]: rename to edge_intersects_triangle ?

File: src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        }
+
+        if( result.second == geode::Position::inside
+            || result.second == geode::Position::edge0
+            || result.second == geode::Position::edge1
+            || result.second == geode::Position::edge2
+            || result.first == geode::Position::parallel )
+        {
+            return true;
+        }
```
Comment (MelchiorSchuh) [inline src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp:145]: Are there cases where segment_triangle_inter return something else than outside and there are no intersections ?
If yes, leave as is, if not, leave only return true;

**Review by MelchiorSchuh — COMMENTED** [review]
Should we add a specific binary for surface / curve inspection, or leave those already available and have people use the inspection of these meshes through the Model inspection binary ?

**Review by MelchiorSchuh — APPROVED** [review]
Just add the test to python bindings and the PR is fine by me

**Pull request merged**